pkg/kubelet/server/server.go

```diff
@@ -941,15 +939,21 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 
 	longRunning := strconv.FormatBool(isLongRunningRequest(path))
 
-	servermetrics.HTTPRequests.WithLabelValues(method, path, serverType, longRunning).Inc()
-
 	servermetrics.HTTPInflightRequests.WithLabelValues(method, path, serverType, longRunning).Inc()
 	defer servermetrics.HTTPInflightRequests.WithLabelValues(method, path, serverType, longRunning).Dec()
 
-	startTime := time.Now()
-	defer servermetrics.HTTPRequestsDuration.WithLabelValues(method, path, serverType, longRunning).Observe(servermetrics.SinceInSeconds(startTime))
+	var statusCode string
+	handler := httplog.WithLogging(s.restfulCont, statusesNoTracePred, func(req *http.Request, status int, hijacked bool) {
+		statusCode = strconv.Itoa(status)
+		if hijacked {
+			statusCode = "-1"
```

Comment:
There's a chance this introduces a data race in timeout scenarios where the inner handler is still processing and calls this callback at the same time the timeout filter returns from ServeHTTP early because the request's context expired. The write to `statusCode` here and the read on lines 955-956 could race.